.github/workflows/main.yaml

```diff
@@ -0,0 +1,54 @@
+name: CI
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+
+jobs:
+  lint-test:
+    uses: ./.github/workflows/lint-test.yaml
+
+
+  generate-sha-tag:
+    if: github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    outputs:
+      sha-tag: ${{ steps.sha-tag.outputs.sha-tag }}
+    steps:
+      - name: Create SHA Container tag
+        id: sha-tag
+        run: |
+          tag=$(cut -c 1-7 <<< $GITHUB_SHA)
+          echo "sha-tag=$tag" >> $GITHUB_OUTPUT
+          
+  publish-static-preview:
+    uses: ./.github/workflows/static-preview.yaml
+    with:
+      sha-tag: ${{ needs.generate-sha-tag.outputs.sha-tag }}
+
+
+  build-deploy:
+    if: github.ref == 'refs/heads/main'
+    uses: ./.github/worfklows/build-deploy.yaml
```

Comment:
```suggestion
    uses: ./.github/workflows/build-deploy.yaml
```

Reply:
I need to work on my typing issues 493a9486f917fc0efb6badc7caf4c803696764d3